src/runtime/mini-prompt-api.js

```diff
@@ -0,0 +1,240 @@
+    this.clientConfigManager_ = deps.clientConfigManager();
+    assert(
+      this.clientConfigManager_,
+      'MiniPromptApi requires an instance of ClientConfigManager.'
+    );
+
+    /** @private @const {!./client-event-manager.ClientEventManager} */
+    this.eventManager_ = deps.eventManager();
+  }
+
+  /**
+   * Does the setup required for the mini prompt's display, including injecting
+   * the mini prompt's css.
+   */
+  init() {
+    const head = this.doc_.getHead();
+    if (!head) {
+      return;
+    }
+
+    const url = '$assets$/swg-mini-prompt.css';
+    const existing = head.querySelector(`link[href="${url}"]`);
+    if (existing) {
+      return;
+    }
+
+    // <link rel="stylesheet" href="..." type="text/css">
+    head.appendChild(
+      createElement(this.doc_.getWin().document, 'link', {
+        'rel': 'stylesheet',
+        'type': 'text/css',
+        'href': url,
+      })
+    );
+  }
+
+  /**
+   * Creates the element and displays it on the page.
+   * @param {{
+   *   autoPromptType: (!AutoPromptType),
+   *   callback: (function()|undefined),
+   * }} options
+   */
+  create(options) {
+    if (
+      options.autoPromptType !== AutoPromptType.CONTRIBUTION &&
+      options.autoPromptType !== AutoPromptType.SUBSCRIPTION
+    ) {
+      return;
+    }
+
+    const theme = this.clientConfigManager_.getTheme();
+    const lang = this.clientConfigManager_.getLanguage();
+    let textContent = '';
+    if (options.autoPromptType === AutoPromptType.CONTRIBUTION) {
+      textContent = msg(CONTRIBUTION_TITLE_LANG_MAP, lang) || '';
+    } else if (options.autoPromptType === AutoPromptType.SUBSCRIPTION) {
+      textContent = msg(SUBSCRIPTION_TITLE_LANG_MAP, lang) || '';
+    }
+
+    // Create all the elements for the mini prompt.
+    /** @const {!Element} */
+    const miniPromptDiv = createElement(this.doc_.getWin().document, 'div', {
+      'role': 'dialog',
+      'lang': lang,
+    });
+    miniPromptDiv.classList.add(`swg-mini-prompt-${theme}`);
+    const titleContainerDiv = createElement(
+      this.doc_.getWin().document,
+      'div',
+      {'role': 'button'}
+    );
+    titleContainerDiv.classList.add('swg-mini-prompt-title-container');
+    titleContainerDiv./*OK*/ innerHTML = TITLE_CONTAINER_DIV_HTML.replace(
+      '/$theme$/g',
+      theme
+    ).replace('$textContent$', textContent);
+    const closeContainerDiv = createElement(
+      this.doc_.getWin().document,
+      'div',
+      {}
+    );
+    closeContainerDiv.classList.add('swg-mini-prompt-close-button-container');
+    closeContainerDiv./*OK*/ innerHTML = CLOSE_CONTAINER_DIV_HTML.replace(
+      '$theme$',
+      theme
+    );
+    miniPromptDiv.appendChild(titleContainerDiv);
+    miniPromptDiv.appendChild(closeContainerDiv);
+    this.doc_.getWin().document.body.appendChild(miniPromptDiv);
+
+    // Handle events and logging for the various sub-components.
+    const clickFun = () => {
```

Comment:
Could you add test coverage for this function?

Reply:
Fixed.